web/packages/teleterm/src/ui/ModalsHost/modals/ReAuthenticate/ReAuthenticate.tsx

```diff
@@ -199,19 +197,30 @@ export const ReAuthenticate: FC<{
   );
 };
 
-const mfaTypeToOption = (mfaType: MfaType): Option<string, string> => {
-  let label: string;
-
-  switch (mfaType) {
-    case 'webauthn':
-      label = 'Hardware Key';
-      break;
-    case 'totp':
-      label = 'Authenticator App';
-      break;
-    default:
-      assertUnreachable(mfaType);
+type AvailableMfaType = Option<MfaType, string>;
```

Comment:
Something like this while keeping the `mfaTypeToOption` function.

```typescript
type MfaType =
  | { kind: 'webauthn' }
  | { kind: 'totp' }
  | { kind: 'sso'; challenge: SSOChallenge };
```

This doesn't make a huge difference in this case, but it's a good practice in general – it makes it so that the state and the business logic is not bound to how it's all going to be presented in the UI. If the Select component in the future decides to change the shape of data it accepts, `ReAuthenticate` wouldn't need to worry about it.

But as I said, it's fine to keep it as it is, this type is used in a single place in a single component and it's not exported anywhere.